### PR TITLE
Add column filter component

### PR DESCRIPTION
**Pull request by FlyersPh9**
- Created column filter component.
- Turned the menu component's background into a mixin.
- Labeled inline inputs only get `padding-right` set if a message is present.
- Button overlays within the text field must be placed within `.iui-message` using `.iui-borderless`.

Screen Shot 2021-04-30 at 4 47 34 PM

### CODE REVIEW

File: src/menu/menu.scss [hunk 1]
```diff
@@ -4,6 +4,98 @@
...
+  @include iui-reset;
+  @include iui-font-family;
+  width: 100%;
+  box-shadow: $iui-elevation-2;
+  @include themed {
+    background-color: t(iui-color-background-1);
+  }
+}
+
+@mixin iui-list-menu {
```
Comment (mayank99) [inline src/menu/menu.scss:17]: If all these styles are added here, they must be removed from somewhere else, right?

File: src/table/column-filter.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      padding-right: $iui-l * 2;
+    }
+
+    > .iui-borderless {
+      position: absolute;
+      right: 0;
+    }
+  }
+
+  .iui-right-align {
```
Comment (mayank99) [inline src/table/column-filter.scss:24]: What happens visually without this class? From the name, it seems like this class should not be explicitly needed.

File: src/table/column-filter.scss [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../menu/index';
+
+@mixin iui-column-filter {
+  @include iui-dropdown;
+  padding: $iui-baseline $iui-sm;
+  width: fit-content;
+
+  .iui-overlay {
```
Comment (mayank99) [inline src/table/column-filter.scss:11]: This classname seems too generic 😕

File: src/table/column-filter.scss [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    position: relative;
+
+    > .iui-input {
+      padding-right: $iui-l * 2;
+    }
+
+    > .iui-borderless {
+      position: absolute;
```
Comment (mayank99) [inline src/table/column-filter.scss:19]: Can we avoid `absolute`?

File: src/table/column-filter.scss [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    > .iui-date-time {
+      display: flex;
+
+      > .iui-input-container > .iui-input {
+        width: $iui-3xl * 2;
+      }
+
+      > .iui-input-container + .iui-input-container {
+        margin-left: $iui-s;
```
Comment (mayank99) [inline src/table/column-filter.scss:37]: Maybe margin-right with `:not(:last-of-type)` for clarity?

Comment (mayank99) [reply src/table/column-filter.scss:37]: Might not work actually 🤔

File: src/table/column-filter.scss (hunk 3 above)
Comment (mayank99) [reply src/table/column-filter.scss:11]: I read your PR description, and it seems like `iui-input-combo` (which is not being used anywhere else) would make more sense here, and then you can get rid of `iui-overlay` completely.

**Review by mayank99 — CHANGES_REQUESTED** [review]

File: src/table/column-filter.scss (hunk 2 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:24]: Without `iui-right-align`:
Screen Shot 2021-04-28 at 3 06 36 PM

Comment (mayank99) [reply src/table/column-filter.scss:24]: Right, so we never not want to right alone. I would just make it default behavior that does not need an extra class.

File: src/table/column-filter.scss (hunk 3 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:11]: My thinking here was `.iui-input-combo` would, for example, place a text field + button side by side (plus some other styling like border radius and padding on the text field).
Screen Shot 2021-04-28 at 3 15 17 PM

Then `.iui-input-combo` with the `.iui-overlay` modifier would allow you to instead place the button overtop of the text field instead of next to it.
Screen Shot 2021-04-28 at 3 17 16 PM

I have not thought far enough ahead for how to tackle something like this however. 🤔
Screen Shot 2021-04-28 at 3 18 02 PM

Comment (mayank99) [reply src/table/column-filter.scss:11]: If this was a borderless button: 
image

Wouldn't that be the same thing as what you described with `iui-overlay`?

File: src/table/column-filter.scss (hunk 5 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:37]: It does work for the margin but I'd have to add an extra class to achieve the width applied to the time text field.

Comment (mayank99) [reply src/table/column-filter.scss:37]: Yeah I guess don't worry about it

File: src/menu/menu.scss (hunk 1 above)
Comment (FlyersPh9) [reply src/menu/menu.scss:17]: All of this styling was originally a part of the mixin iui-menu.  I split it up into 2 mixins so I could reuse the background & drop shadow portion without bringing along the list styling.  GitHub seems to be marking way more content as new than should be.  Here is a before & after comparison in VSC.
Screen Shot 2021-04-28 at 4 26 16 PM

File: src/table/column-filter.scss (hunk 2 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:24]: The toggle switch is still left aligned though.

Comment (mayank99) [reply src/table/column-filter.scss:24]: How does that change anything? You can just right align all `.iui-input-container` elements

File: src/table/column-filter.scss (hunk 3 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:11]: This one's complicated to explain.  The borderless button overlaying the text field should only ever display an icon, no text.  I have a fixed padding so that the text field value doesn't clash with the borderless button.  You can see the padding in the image below.
Screen Shot 2021-04-28 at 4 39 47 PM

If I swap the borderless button with the high-visibility and swap the icon with text, the text field's value will go underneath the button.
Screen Shot 2021-04-28 at 4 43 55 PM

At first this seems ok, but you can not see the last few characters of whatever is typed into the text field.  The GIF below demos this well, the text inside is "This is the column filter in iTwinUI" but you can never see "UI" because the button covers it up.  I think this would be even worse if the button was in the front rather than at the rear.
Screen Recording 2021-04-28 at 4 49 02 PM

I thought about reversing it too, but if we place all buttons outside, then the borderless button will look out of place.
Screen Shot 2021-04-28 at 4 54 52 PM

File: src/table/column-filter.scss (hunk 4 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:19]: I don't know how else to position the calendar icon / borderless button right aligned within the text field.  I thought about exploring something like below, but seems like a bad idea:
```
<input>
  <button />
</input>
```

Comment (mayank99) [reply src/table/column-filter.scss:19]: Should be possible to do with flex but could get complicated 🤔

I guess it's fine for this component but when we add the dedicated combo input component, I would expect a proper solution.

File: src/inputs/labeled-inputs.scss [hunk 6]
```diff
@@ -106,15 +106,19 @@
...
-    > .iui-input,
-    > .iui-textarea {
-      padding-right: $iui-xl;
-
-      + .iui-message {
-        position: absolute;
-        top: 50%;
-        right: $iui-sm;
-        transform: translateY(-50%);
+    &.iui-positive,
```
Comment (bentleyvk) [inline src/inputs/labeled-inputs.scss:109]: Why this change is needed?

File: src/table/column-filter.scss (hunk 4 above)
Comment (bentleyvk) [reply src/table/column-filter.scss:19]: This can be achieved much simpler, I think:
image
And also you need to revert changes in `labeled-inputs.scss`:
image
Result:
image

File: src/table/column-filter.scss (hunk 2 above)
Comment (bentleyvk) [reply src/table/column-filter.scss:26]: Yeah, seems like everything should be aligned to the right be default.

File: src/table/column-filter.scss [hunk 7]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../menu/index';
+
+@mixin iui-column-filter {
+  @include iui-dropdown;
+  padding: $iui-baseline $iui-sm;
+  width: fit-content;
```
Comment (bentleyvk) [inline src/table/column-filter.scss:9]: `width: fit-content` is not supported by IE. You can achieve the same with:
image
(I don't know why width is crossed but it is needed.)

File: src/table/column-filter.scss [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+  .iui-right-align {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-end;
+
+    > .iui-date-time {
+      display: flex;
+
+      > .iui-input-container > .iui-input {
```
Comment (bentleyvk) [inline src/table/column-filter.scss:34]: Do we need this one?
image
It is very wide and for me this one is enough when we add time selection to date picker:
image

File: src/table/column-filter.scss [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+      border-bottom-left-radius: 0;
+    }
+  }
+
+  .iui-right-align {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-end;
+
+    > .iui-date-time {
```
Comment (bentleyvk) [inline src/table/column-filter.scss:31]: I see it is not applied to all inputs in example.

File: src/menu/menu.scss (hunk 1 above)
Comment (bentleyvk) [reply src/menu/menu.scss:17]: Ok, I don't understand why `iui-list-menu` is needed because it almost the same as `iui-menu`. Are you sure you don't forgot to delete one or the other?
menu_mixin

File: src/menu/menu.scss [hunk 10]
```diff
@@ -4,6 +4,98 @@
 @import '../inputs/variables';
 @import '../icon/index';
 
+@mixin iui-dropdown {
+  @include iui-reset;
+  @include iui-font-family;
+  width: 100%;
+  box-shadow: $iui-elevation-2;
```
Comment (bentleyvk) [inline src/menu/menu.scss:11]: Do we really need to reuse stuff? Because the only reusable part is this `box-shadow` and `background-color` which are kind of already defined as a standard.

File: src/table/column-filter.scss [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+      }
+    }
+  }
+
+  > .iui-button-bar {
+    margin-top: $iui-baseline;
+    display: flex;
+    align-items: center;
```
Comment (bentleyvk) [inline src/table/column-filter.scss:52]: This is not really needed.

File: src/menu/menu.scss (hunk 1 above)
Comment (veekeys) [reply src/menu/menu.scss:17]: Yeah, its like a duplicate again...

File: src/menu/menu.scss (hunk 10 above)
Comment (veekeys) [reply src/menu/menu.scss:11]: Maybe it is ok to have that mixin to share that basic stuff... But I do agree with @bentleyvk here too, as its only 2 rules here.. 
If we start making mixins for every duplication of 2 rules, might end up in a huge mess :D

File: src/menu/menu.scss (hunk 1 above)
Comment (FlyersPh9) [reply src/menu/menu.scss:17]: You are correct, I forgot to delete the duplicate. 😅. Fixed in my last push.

Comment (mayank99) [reply src/menu/menu.scss:17]: There you go 😄

File: src/table/column-filter.scss (hunk 2 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss]: Latest push right aligns everything & we explicitly tell the toggle switch to be left aligned.

File: src/inputs/labeled-inputs.scss (hunk 6 above)
Comment (FlyersPh9) [reply src/inputs/labeled-inputs.scss:109]: I was encountering this issue.  A narrow, inline text field is reserving the space for a status icon & encroaching on the text.
Screen Shot 2021-04-29 at 9 58 39 AM

File: src/table/column-filter.scss (hunk 8 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss]: I can remove it, will certainly simplify the CSS.  We will need to update the standard documentation to remove the wider option.  That option was implemented before we had a date + time combo picker.

Comment (FlyersPh9) [reply src/table/column-filter.scss]: It's been removed.

File: src/table/column-filter.scss (hunk 9 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss]: This has been removed in response to [this discussion](https://github.com/iTwin/iTwinUI/pull/51#discussion_r622831663).

File: src/menu/menu.scss (hunk 10 above)
Comment (FlyersPh9) [reply src/menu/menu.scss:11]: I like the idea of a central location for all dropdown menu styling.  If we ever decided to change the styling, we could change it one place and have it propagate across all instances.  But I do understand the concern of introducing too many mixins.

File: src/table/column-filter.scss (hunk 11 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss]: Removed the unnecessary code.

File: src/table/column-filter.scss (hunk 7 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:9]: If I remove `width: fit-content;` I get uneven text fields.
Screen Shot 2021-04-29 at 11 06 18 AM

So then I tried applying `width: 100%;` to `.iui-input-container` and that messes up the other filter options.
Screen Shot 2021-04-29 at 11 07 26 AM

Unsure how to fix this at the moment. 🤔

Comment (FlyersPh9) [reply src/table/column-filter.scss]: Figured it out! I can apply `width: 100%` to `.iui-input-container:not(.iui-inline)`.

**Comment by FlyersPh9** [comment]
@bentleyvk The tests on the column filters are giving results I didn't expect.  I tried to leave a comment directly on [this image](https://github.com/iTwin/iTwinUI/pull/51/files?short_path=a20cdce#diff-a20cdced083d30126996b097c50410cc35851c03d3c2eb7f01911cb9513b2003), but I couldn't find the option in GitHub.

I applied 16px padding to the `<section>` so that we could capture the drop shadow effect.  I'm not sure why, but the right side seems to get cutoff. I approved the tests as is because I can confirm that the results are still correct, but thought I would ask if you know of a solution?
Screen Shot 2021-04-29 at 11 52 28 AM
Screen Shot 2021-04-29 at 11 52 47 AM

File: src/menu/menu.scss [hunk 12]
```diff
@@ -4,19 +4,23 @@
 @import '../inputs/variables';
 @import '../icon/index';
 
-@mixin iui-menu {
+@mixin iui-dropdown {
   @include iui-reset;
   @include iui-font-family;
   width: 100%;
```
Comment (mayank99) [inline src/menu/menu.scss:10]: I think this is causing problems. It's taking extra width that it's not supposed to, which somehow makes it wider than the section itself.

image

File: src/table/column-filter.scss (hunk 4 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:26]: That's a good suggestion, I didn't think to try that.

However, I was trying to set this part up for a future component, which I think would need to handle the icon placement differently.  I have [a branch where I did some basic testing](https://github.com/iTwin/iTwinUI/compare/main...jon/input-combinations).  Below is a screenshot from that branch where I'm trying to throw every option at it.
Screen Shot 2021-04-29 at 12 12 12 PM

Your solution would work in the column filter, but would not work if `.iui-inline` is not applied.

File: src/menu/menu.scss (hunk 12 above)
Comment (FlyersPh9) [reply src/menu/menu.scss]: Good catch!  I've moved `width: 100%;` from the dropdown mixin to the list-menu mixin.

Comment (mayank99) [reply src/menu/menu.scss]: Don't forget to update your comment above so Vilius knows it's fixed 😄

**Review by mayank99 — APPROVED** [review]
Looks ok in current state.

The only thing we need to think about is how to handle date input in react. We are not using `<input type='date'>` which handles a lot of things for us.
image

File: src/table/column-filter.scss (hunk 3 above)
Comment (mayank99) [reply src/table/column-filter.scss]: Leaving this open because I still don't like `iui-icon-overlay` class 😄

**Comment by bentleyvk** [comment]
> Looks ok in current state.
> 
> The only thing we need to think about is how to handle date input in react. We are not using `<input type='date'>` which handles a lot of things for us.
> image

I think we really can't use it because if I am not mistaken it will open native date picker in modern browsers. In other project I was using `date-fns` to parse and validate the date string that was written by hand.

File: src/table/column-filter.scss (hunk 3 above)
Comment (bentleyvk) [reply src/table/column-filter.scss]: I don't like it, too. Because we won't be able to reuse Input component in React. We will need to enter whole input HTML just to wrap input with button in `iui-icon-overlay` class.

File: src/table/column-filter.scss (hunk 4 above)
Comment (bentleyvk) [reply src/table/column-filter.scss:26]: If @mayank99 doesn't mind I will unresolve this conversation. If we want to achieve this combo with button and input then I think we don't need to use `position: absolute` but to have the same as button group. But then I don't think borderless button fits here as it is inside of the input field, not next to it as in your example in previous comment.
My recommendation would be maybe going with my example for now and when we implement combo input with button, we can refactor it. Because currently people are waiting for table filtering and some of them are not switching to itwinui from bwc because of this.

File: src/inputs/labeled-inputs.scss (hunk 6 above)
Comment (bentleyvk) [reply src/inputs/labeled-inputs.scss:109]: If you decide to go with my suggestion, instead of adding status classes you could do the same trick you did in Header component - checking if input is the last element and only then applying padding.

File: src/menu/menu.scss [hunk 13]
```diff
@@ -71,7 +75,7 @@
 
       > .iui-icon {
         @include themed {
-          fill: t(iui-icons-color-primary);
+          fill: t(iui-color-foreground-primary);
```
Comment (bentleyvk) [inline src/menu/menu.scss:78]: Are these changes intended? Or they were done before using icon colors everywhere and now left?

File: src/table/column-filter.scss [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    > .iui-borderless {
+      position: absolute;
+      right: 0;
+      border-top-left-radius: 0;
+      border-bottom-left-radius: 0;
+    }
+  }
+
+  > .iui-input-container {
+    align-self: flex-end;
```
Comment (bentleyvk) [inline src/table/column-filter.scss:34]: Is this needed? Because toggling it seems to do nothing.

File: backstop/tests/column-filter.html [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+    </style>
+    <script src="../assets/theme.js"></script>
+  </head>
+  <body class="iui-body">
+    <theme-button></theme-button>
+    <h1>Column filter</h1>
+    <hr />
+    <h2 id="default">Text filter</h2>
+
+    <section id="demo-text">
```
Comment (bentleyvk) [inline backstop/tests/column-filter.html:50]: Without `<section>` that has `fit-content` filter takes whole screen size. Our filter will be placed inside `<div>`, so it will also take whole screen.
Change `display: flex` to `display: inline-flex` in `iui-column-filter`.

File: backstop/tests/select.html [hunk 16]
```diff
@@ -14,6 +14,10 @@
       rel="stylesheet"
       href="../../lib/css/global.css"
     />
+    <link
+      rel="stylesheet"
+      href="../../lib/css/menu.css"
```
Comment (bentleyvk) [inline backstop/tests/select.html:19]: Is it needed?

File: src/table/column-filter.scss (hunk 4 above)
Comment (veekeys) [reply src/table/column-filter.scss:26]: We should be able to reuse inline input. We already have this covered, we just need to put a borderless icon instead of simple svg under `.iui-message`.
I have tried doing that in storybook (instead of svg icon, I passed IconButton) and got this one:
image

So this is pretty close what we need, IMO. Just overriding `right` property value with 0 does the trick.
image

From React side I think that would be pretty comfortable to do instead of having fully custom HTML.

**Comment by veekeys** [comment]
@FlyersPh9 Is this toggle really supposed to be on the left? I just look at it and cannot get used to..
image

What if we put it aligned all together?
image

File: src/table/column-filter.scss (hunk 4 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss]: Followed the advice suggested here, I responded in [a different convo here](https://github.com/iTwin/iTwinUI/pull/51#discussion_r624160174).

File: src/inputs/labeled-inputs.scss (hunk 6 above)
Comment (FlyersPh9) [reply src/inputs/labeled-inputs.scss]: The padding is no longer applied based on positive, warning, or negative classes.  Instead we always apply the extra padding, but if the input is the last element (meaning there is no status message provided) then we remove the extra padding.

I am still have a padding issue for non-inline inputs, [see the convo here](https://github.com/iTwin/iTwinUI/pull/51#discussion_r624160174).

**Comment by FlyersPh9** [comment]
> > Looks ok in current state.
> > The only thing we need to think about is how to handle date input in react. We are not using `<input type='date'>` which handles a lot of things for us.
> > image
> 
> I think we really can't use it because if I am not mistaken it will open native date picker in modern browsers. In other project I was using `date-fns` to parse and validate the date string that was written by hand.

This will let us remove the built-in icon.  [Source.](https://stackoverflow.com/questions/11320615/disable-native-datepicker-in-google-chrome)
```
&[type='date'] {
    &::input-placeholder { 
        visibility: hidden !important;
    }
    &::-webkit-calendar-picker-indicator {
        display: none;
    }
}
```
MicrosoftTeams-image (2)

File: src/menu/menu.scss (hunk 13 above)
Comment (FlyersPh9) [reply src/menu/menu.scss:78]: I think these are from before the icon color change PR.

File: src/table/column-filter.scss (hunk 14 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:18]: `iui-input-container` is no more 🪦

File: src/table/column-filter.scss (hunk 3 above)
Comment (mayank99) [reply src/table/column-filter.scss]: We could probably handle the non-inline case in the future

File: backstop/tests/column-filter.html (hunk 15 above)
Comment (FlyersPh9) [reply backstop/tests/column-filter.html:50]: > @FlyersPh9 Is this toggle really supposed to be on the left? I just look at it and cannot get used to..
> image
> 
> What if we put it aligned all together?
> image

That's how the standard has it, I believe these were originally laid out by JP.  One thing I thought about proposing was something like the attached Photoshop mockup where the toggle is left aligned but on the same line as the buttons.
Screen Shot 2021-04-30 at 4 15 40 PM

File: src/table/column-filter.scss (hunk 14 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss]: You are correct, it seems like it is not needed.  Removed.

File: backstop/tests/column-filter.html (hunk 15 above)
Comment (FlyersPh9) [reply backstop/tests/column-filter.html:50]: > Change `display: flex` to `display: inline-flex` in `iui-column-filter`.

I don't know why GitHub put that other response in here 😅 But this change is now implemented.

File: src/table/column-filter.scss [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+        margin-left: auto;
+      }
+
+      &:not(:last-child) {
+        margin-right: $iui-s;
+      }
+    }
+
+    > .iui-toggle-switch {
+      display: inline-flex;
```
Comment (bentleyvk) [inline src/table/column-filter.scss:37]: It looks like it does nothing.

File: src/menu/menu.scss (hunk 13 above)
Comment (bentleyvk) [reply src/menu/menu.scss:78]: So shouldn't we change that back to use icon colors? 😀

File: src/inputs/labeled-inputs.scss [hunk 18]
```diff
@@ -8,6 +8,7 @@
   display: flex;
   flex-direction: column;
   flex-wrap: nowrap;
+  position: relative;
```
Comment (bentleyvk) [inline src/inputs/labeled-inputs.scss:11]: Inline style has this rule already:
image

File: src/inputs/labeled-inputs.scss [hunk 19]
```diff
@@ -52,26 +53,36 @@
...
     line-height: $iui-line-height;
     margin-top: $iui-inputs-margin;
-
     @include themed {
       color: t(iui-text-color-muted);
     }
 
-    svg {
+    > svg,
+    :where(:not(.iui-button)) svg {
```
Comment (bentleyvk) [inline src/inputs/labeled-inputs.scss:61]: I don't think `:where` is needed here.

File: backstop/tests/column-filter.html [hunk 20]
```diff
@@ -0,0 +1,221 @@
...
+      rel="stylesheet"
+      href="../../lib/css/inputs.css"
+    />
+    <link
+      rel="stylesheet"
+      href="../../lib/css/toggle-switch.css"
+    />
+    <link
+      rel="stylesheet"
+      href="../../lib/css/table.css"
```
Comment (veekeys) [inline backstop/tests/column-filter.html:31]: I am just thinking now... If column filter is going to be a separate component, we might want to import only filtering css, not whole table css. @bentleyvk thoughts?

File: src/inputs/labeled-inputs.scss [hunk 21]
```diff
@@ -52,26 +53,36 @@
...
+    }
+
     a {
       user-select: none;
     }
   }
 
+  &:not(.iui-inline) > .iui-message > .iui-borderless {
+    position: absolute;
+    top: $iui-l + 1;
```
Comment (bentleyvk) [inline src/inputs/labeled-inputs.scss:82]: Try centering like inline:
image

File: src/inputs/labeled-inputs.scss [hunk 22]
```diff
@@ -101,20 +112,30 @@
...
+        right: 0;
         transform: translateY(-50%);
+
+        > svg,
+        :where(:not(.iui-button)) svg {
+          margin-right: $iui-sm;
+        }
+      }
+
+      &:last-child {
```
Comment (veekeys) [inline src/inputs/labeled-inputs.scss:137]: I guess you could use nth-child selector for non inline input to achieve same, if you wanted...

File: src/table/column-filter.scss (hunk 17 above)
Comment (veekeys) [inline src/table/column-filter.scss:37]: Is it needed?

File: src/menu/menu.scss (hunk 13 above)
Comment (FlyersPh9) [reply src/menu/menu.scss]: I've updated all the icon fills to use the icon variables but only within `menu.scss`.

File: src/table/column-filter.scss (hunk 17 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:37]: I think you are right, removed.

File: src/inputs/labeled-inputs.scss (hunk 18 above)
Comment (FlyersPh9) [reply src/inputs/labeled-inputs.scss:11]: Removed `position: relative` from `.iui-inline` and now only have it in the root of the SCSS file.

File: src/inputs/labeled-inputs.scss (hunk 19 above)
Comment (FlyersPh9) [reply src/inputs/labeled-inputs.scss:61]: Replaced with `:is(:not(.iui-button)) svg`, credit goes to @mayank99 for finding this one.

File: src/table/column-filter.scss (hunk 17 above)
Comment (FlyersPh9) [reply src/table/column-filter.scss:37]: [It's been removed.](https://github.com/iTwin/iTwinUI/pull/51#discussion_r625334727)

File: src/inputs/labeled-inputs.scss (hunk 21 above)
Comment (FlyersPh9) [reply src/inputs/labeled-inputs.scss:82]: Good suggestion, implemented.

Comment (mayank99) [reply src/inputs/labeled-inputs.scss]: It won't work for non-inline
image

But we've removed the non-inline example from html (for now) so maybe remove it here from css too

File: src/inputs/labeled-inputs.scss (hunk 22 above)
Comment (mayank99) [reply src/inputs/labeled-inputs.scss:137]: This is what @FlyersPh9 said to me:

> the borderless button is within the message class.
> 
> ```
> <label/>
> <input/>
> <message>
>   <button>
> </message>
> ```
> 
> so i cant think of a way to apply padding to the input based on if a button is present within message. 🤔

I don't see how `nth-child` would help with that. The html structure is the same.

I think this is why we need to avoid `absolute` positioning, and try to use flex instead. But that's for future I guess.

File: backstop/tests/column-filter.html (hunk 20 above)
Comment (bentleyvk) [reply backstop/tests/column-filter.html:31]: It kind of makes sense to have everything table related under table CSS. But it takes a huge table CSS file. However if you are using table those rules are going to be imported anyway 😀 I wonder if anyone wants to use filters outside the table. But maybe we can still have it as a separate component as it already has its own test file.

File: src/inputs/labeled-inputs.scss (hunk 21 above)
Comment (bentleyvk) [reply src/inputs/labeled-inputs.scss]: Maybe yeah, remove it for now and finish it up later with input + button combo.

File: src/inputs/labeled-inputs.scss (hunk 22 above)
Comment (veekeys) [reply src/inputs/labeled-inputs.scss:137]: Oh yeah, now I see.. the html is just wrong.. We cannot have message container, where part of it is under input, part of it is in input.. It is just wrong, where elements are magically appearing on totally different places. 
We might need to refactor this thing... again... Cause we already saw some use cases, where this format just does not work. We might just want separate sections tackling with css:
label above input
input
right input icon
message below.

File: src/inputs/labeled-inputs.scss [hunk 23]
```diff
@@ -52,26 +53,37 @@
...
+      border-top-left-radius: 0;
+      border-bottom-left-radius: 0;
+    }
+
     a {
       user-select: none;
     }
   }
 
+  &:not(.iui-inline) > .iui-message > .iui-borderless {
```
Comment (veekeys) [inline src/inputs/labeled-inputs.scss:80]: So this one is not needed then as you removed non inline example.

File: src/menu/menu.scss [hunk 24]
```diff
@@ -85,7 +89,7 @@
   
       > .iui-icon {
         @include themed {
-          fill: t(iui-icons-color-actionable-disabled);
+          fill: t(iui-icons-color);
```
Comment (bentleyvk) [inline src/menu/menu.scss:92]: Why still different color when it was before?

File: src/inputs/labeled-inputs.scss (hunk 21 above)
Comment (FlyersPh9) [reply src/inputs/labeled-inputs.scss]: Reverted back to `top: $iui-l + 1;`.

File: src/menu/menu.scss (hunk 24 above)
Comment (FlyersPh9) [reply src/menu/menu.scss:92]: Someone must have screwed up 👀 It is fixed now.

File: src/inputs/labeled-inputs.scss (hunk 23 above)
Comment (FlyersPh9) [reply src/inputs/labeled-inputs.scss]: Removed.

**Review by bentleyvk — APPROVED** [review]

**Review by mayank99 — APPROVED** [review]
